src/hotspot/share/memory/memoryReserver.cpp

```diff
@@ -546,16 +546,16 @@ ReservedHeapSpace HeapReserver::Instance::reserve_compressed_oops_heap(const siz
 
   const size_t attach_point_alignment = lcm(alignment, os_attach_point_alignment);
 
-  char* aligned_heap_base_min_address = align_up((char*)HeapBaseMinAddress, alignment);
-  size_t noaccess_prefix = ((aligned_heap_base_min_address + size) > (char*)OopEncodingHeapMax) ?
+  uintptr_t aligned_heap_base_min_address = align_up(MAX2(HeapBaseMinAddress, alignment), alignment);
```

Comment:
Just to be clear, this is the crux of the fix, where we ensure the min-address is now never zero - right?